File: example/src/androidTest/java/com/github/bassaer/example/matchers/TextColorMatcher.java
package com.github.bassaer.example.matchers;

import android.view.View;
import android.widget.TextView;

import org.hamcrest.Description;
import org.hamcrest.TypeSafeMatcher;

/**
 * Custom matcher to test TextView color
 * Created by nakayama on 2017/08/02.
 */

public class TextColorMatcher extends TypeSafeMatcher<View> {
    private int mExpectedColor = -1;

    public TextColorMatcher(int color) {
        mExpectedColor = color;
    }

    @Override
    protected boolean matchesSafely(View view) {
        TextView textView = (TextView)view;
        return mExpectedColor == textView.getCurrentTextColor();
    }

    @Override
    public void describeTo(Description description) {
        description.appendText("with color value: ");
        description.appendValue(String.valueOf(mExpectedColor));
    }


}

File: example/src/androidTest/kotlin/com/github/bassaer/example/matcher/TextColorMatcher.kt
package com.github.bassaer.example.matcher

import android.view.View
import android.widget.TextView

import org.hamcrest.Description
import org.hamcrest.TypeSafeMatcher

/**
 * Custom matcher to test TextView color
 * Created by nakayama on 2017/08/02.
 */

class TextColorMatcher(color: Int) : TypeSafeMatcher<View>() {
    private var mExpectedColor = -1

    init {
        mExpectedColor = color
    }

    override fun matchesSafely(view: View): Boolean {
        val textView = view as TextView
        return mExpectedColor == textView.currentTextColor
    }

    override fun describeTo(description: Description) {
        description.appendText("with color value: ")
        description.appendValue(mExpectedColor.toString())
    }
}
